code/modules/mob/living/simple_animal/slime/slime.dm

```diff
@@ -199,7 +200,9 @@
 /mob/living/simple_animal/slime/proc/random_colour()
 	set_colour(pick(slime_colours))
 
-/mob/living/simple_animal/slime/regenerate_icons()
+/mob/living/simple_animal/slime/regenerate_icons(force_mood)
+	if(force_mood)
+		current_mood = force_mood
```

Comment:
Don't add extra parameters to procs like this. Have a separate proc called `force_mood()` that sets the mood then calls regenerate_icons.